circuits/benches/internal_match_settlement.rs

```diff
@@ -0,0 +1,195 @@
+//! Benchmarks internal match settlement
+#![allow(incomplete_features)]
+#![allow(missing_docs)]
+#![feature(generic_const_exprs)]
+
+use circuit_types::traits::{CircuitBaseType, SingleProverCircuit};
+use circuit_types::PlonkCircuit;
+use circuits::zk_circuits::valid_match_settle::test_helpers::dummy_witness_and_statement;
+use circuits::zk_circuits::valid_match_settle::{
+    ValidMatchSettle, ValidMatchSettleStatement, ValidMatchSettleWitness,
+};
+use circuits::{singleprover_prove, verify_singleprover_proof};
+use constants::{MAX_BALANCES, MAX_ORDERS};
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use mpc_relation::proof_linking::LinkableCircuit;
+
+/// The parameter set for the small sized circuit (MAX_BALANCES, MAX_ORDERS,
+/// MERKLE_HEIGHT)
+const SMALL_PARAM_SET: (usize, usize) = (2, 2);
+/// The parameter set for the large sized circuit
+const LARGE_PARAM_SET: (usize, usize) = (MAX_BALANCES, MAX_ORDERS);
+
+// -----------
+// | Helpers |
+// -----------
+
+/// Create a witness and statement with the given sizing generics
+pub fn create_sized_witness_statement<const MAX_BALANCES: usize, const MAX_ORDERS: usize>() -> (
+    ValidMatchSettleWitness<MAX_BALANCES, MAX_ORDERS>,
+    ValidMatchSettleStatement<MAX_BALANCES, MAX_ORDERS>,
+)
+where
+    [(); MAX_BALANCES + MAX_ORDERS]: Sized,
+{
+    dummy_witness_and_statement::<MAX_BALANCES, MAX_ORDERS>()
+}
```

Comment:
Is this helper necessary? it just directly calls `dummy_witness_and_statement` with the same generics

Reply:
good call